packages/@react-spectrum/combobox/src/MobileComboBox.tsx

```diff
@@ -55,20 +56,39 @@ export const MobileComboBox = React.forwardRef(function MobileComboBox<T extends
     isQuiet,
     isDisabled,
     validationState,
-    isReadOnly
+    isReadOnly,
+    menuTrigger = 'input'
   } = props;
 
   let {contains} = useFilter({sensitivity: 'base'});
   let state = useComboBoxState({
     ...props,
     defaultFilter: contains,
     allowsEmptyCollection: true,
-    shouldCloseOnBlur: false
+    shouldCloseOnBlur: true
   });
 
   let buttonRef = useRef<HTMLElement>();
   let domRef = useFocusableRef(ref, buttonRef);
   let {triggerProps, overlayProps} = useOverlayTrigger({type: 'listbox'}, state, buttonRef);
```

Comment:
Note that I haven't fixed the following bug yet: `Can't close "menuTrigger='focus'" combobox on mobile unless you select an option`. The core issue is that closing the tray sets `isFocused` to false because the input field loses focus.

One way to fix this would be to restore the following changes from a older pull:
- https://github.com/adobe/react-spectrum/blob/975957a3c3fb4a9f3be358428e0980a4d4e51c1a/packages/%40react-aria/combobox/src/useComboBox.ts#L140 
-  https://github.com/adobe/react-spectrum/blob/975957a3c3fb4a9f3be358428e0980a4d4e51c1a/packages/%40react-aria/combobox/src/useComboBox.ts#L230-L246
(for the second bullet point, specifically the logic where we track `prevOpenState` and have modify the `(isMobile && state.isOpen && e.relatedTarget === null)` part of the `onBlur` logic to see if we are going from `prevOpenState = open` to `prevOpenState = close`.

Perhaps another way would be to track `prevOpenState` in `useComboBoxState` and block `setFocused` from triggering `open`. Maybe we could just have isFocused not become false when the user closes the tray...

Reply:
e9d81f9 is what I settled on to fix this, open to suggestions